src/components/Table/RowState.ts

```diff
@@ -80,23 +80,26 @@ export class RowState {
   }
 
   loadSelected(rows: GridDataRow<any>[]): void {
-    const selectedRows = rows.filter(row => row.initSelected)
-    // Initialize with selected rows as defined 
+    const selectedRows = rows.filter((row) => row.initSelected);
+    // Initialize with selected rows as defined
     const map = new Map<string, SelectedState>();
-    selectedRows.forEach(row => {
+    selectedRows.forEach((row) => {
       map.set(row.id, "checked");
-    })
+    });
 
     this.selectedRows.merge(map);
   }
 
   // Updates the list of rows and regenerates the collapsedRows property if needed.
   setRows(rows: GridDataRow<any>[]): void {
-    // If the set of rows are different, and this is the first time adding "data" rows (non "totals" or "header" rows), then collapsedRows may need to be updated.
-    if (rows !== this.rows && this.rows.some((r) => r.kind !== "totals" && r.kind !== "header")) {
```

Comment:
Fwiw @bdow I was able to get the new test to pass by just changing this line to:

```
    if (rows !== this.rows && rows.some((r) => r.kind !== "totals" && r.kind !== "header")) {
```

I.e. checking the _incoming_ rows for "we have a real row" now.

Previously both `setRows` calls from the test were not entering this block b/c

* First call, this.rows is empty
* Second call, this.rows is the original static total+header

...I guess your new line 102 does add a `!` that was not here before:

```
    if (rows !== this.rows && !this.rows.some((r) => r.kind !== "totals" && r.kind !== "header")) {
```

So perhaps that is the core fix? If so, that makes sense; given the lack of `persistCollapse` in the test and just not 100% following line 116, I wonder if the `checkLocalStorage` aspect of this code either wasn't necessary and/or isn't being covered...